### ISSUE TITLE
[WebtoolsE2E][Aspire] Failed to click on any instrument in the Aspire dashboard "Metrics" with some errors "Error: System.InvalidOperationException: Object of type 'Aspire.Dashboard.Components.ChartFilters' does not have a property matching the name 'ViewModel'."

### ISSUE THREAD

**Issue by Rita003** [dotnet/aspire#4937]
**REGRESSION INFO**: Work in Aspire 8.0.2

**INSTALL STEPS**

1. Clean machine: Win11 x64 23h2 ENU
2. Install SDK 8.0.303
3. Install Aspire 8.1.0-preview.1.24366.1
4. Apply feed

**REPRO STEPS**

1. Open CMD, create an Aspire Starter App using following commands:

```
dotnet new aspire-starter -o AspireStarterApp
cd AspireStarterApp/AspireStarterApp.AppHost
```
2. Dotnet run the project in CMD, and open URL in the browser.
`dotnet run`
3. Click the "Metrics" option on the Aspire dashboard left, click the "Select a resource" and select the apiservice(or webfrontend) option, then click on any instrument.

**ACTUAL**
The page shows "An unhandled error has occurred. Reload" with some errors in the console.
![Screenshot 2024-07-17 134716](https://github.com/user-attachments/assets/6a02b3e1-5348-4c2e-abe2-0a94d5cb26d1)
There are some errors in the CMD
![Screenshot 2024-07-17 134826](https://github.com/user-attachments/assets/64705b4a-68e0-4198-ae87-7205f0038cf6)

**EXPECTED**
There is no errors occurred when click on any instrument in the Metrics

**Error Log**
**Console Log**:
```
blazor.web.js:1  [2024-07-17T05:45:41.158Z] Error: System.InvalidOperationException: Object of type 'Aspire.Dashboard.Components.ChartFilters' does not have a property matching the name 'ViewModel'.
   at Microsoft.AspNetCore.Components.Reflection.ComponentProperties.ThrowForUnknownIncomingParameterName(Type targetType, String parameterName)
   at Microsoft.AspNetCore.Components.Reflection.ComponentProperties.SetProperties(ParameterView& parameters, Object target)
   at Microsoft.AspNetCore.Components.ComponentBase.SetParametersAsync(ParameterView parameters)
   at Microsoft.AspNetCore.Components.Rendering.ComponentState.SupplyCombinedParameters(ParameterView directAndCascadingParameters)
log @ blazor.web.js:1
blazor.web.js:1 [2024-07-17T05:45:41.159Z] Information: Connection disconnected.
blazor.web.js:1  Uncaught (in promise) Error: Cannot send data if the connection is not in the 'Connected' State.
    at Nn.send (blazor.web.js:1:85140)
    at gn._sendMessage (blazor.web.js:1:58610)
    at gn._sendWithProtocol (blazor.web.js:1:58700)
    at gn.send (blazor.web.js:1:58808)
    at Fo.beginInvokeDotNetFromJS (blazor.web.js:1:139418)
    at y.invokeDotNetMethodAsync (blazor.web.js:1:4125)
    at S.invokeMethodAsync (blazor.web.js:1:5633)
    at app.js:295:22
    at later (app.js:287:22)
[NEW] Explain Console errors by using Copilot in Edge: click to explain an error. 
```
**CMD Log**
```
fail: Aspire.Hosting.Dashboard.Microsoft.AspNetCore.Components.Server.Circuits.CircuitHost[111]
      Unhandled exception in circuit 'LCezhu9nGQZ_RonKWEdMXK5gmHbJoFrvDJxDeCX7zww'.
      System.InvalidOperationException: Object of type 'Aspire.Dashboard.Components.ChartFilters' does not have a property matching the name 'ViewModel'.
         at Microsoft.AspNetCore.Components.Reflection.ComponentProperties.ThrowForUnknownIncomingParameterName(Type targetType, String parameterName)
         at Microsoft.AspNetCore.Components.Reflection.ComponentProperties.SetProperties(ParameterView& parameters, Object target)
         at Microsoft.AspNetCore.Components.ComponentBase.SetParametersAsync(ParameterView parameters)
         at Microsoft.AspNetCore.Components.Rendering.ComponentState.SupplyCombinedParameters(ParameterView directAndCascadingParameters)
fail: Aspire.Hosting.Dashboard.Microsoft.AspNetCore.Components.Server.Circuits.CircuitHost[111]
      Unhandled exception in circuit 'LCezhu9nGQZ_RonKWEdMXK5gmHbJoFrvDJxDeCX7zww'.
      System.ObjectDisposedException: Cannot access a disposed object.
         at Microsoft.AspNetCore.Components.RenderTree.RenderTreeDiffBuilder.InsertNewFrame(DiffContext& diffContext, Int32 newFrameIndex)
         at Microsoft.AspNetCore.Components.RenderTree.RenderTreeDiffBuilder.AppendDiffEntriesForRange(DiffContext& diffContext, Int32 oldStartIndex, Int32 oldEndIndexExcl, Int32 newStartIndex, Int32 newEndIndexExcl)
         at Microsoft.AspNetCore.Components.Rendering.ComponentState.RenderIntoBatch(RenderBatchBuilder batchBuilder, RenderFragment renderFragment, Exception& renderFragmentException)
         at Microsoft.AspNetCore.Components.RenderTree.Renderer.ProcessRenderQueue()
```

**Comment by adamint** [comment]
already fixed in #4934

**Comment by Rita003** [comment]
Verify this issue fixed on the latest Aspire 8.1.0-preview.1.24367.1
